core/src/main/java/com/linecorp/armeria/server/healthcheck/HealthCheckServiceBuilder.java

```diff
@@ -256,6 +258,22 @@ public HealthCheckServiceBuilder updatable(HealthCheckUpdateHandler updateHandle
         return this;
     }
 
+    /**
+     * Adds a {@link HealthCheckUpdateHandler} which invoked when the healthiness of the {@link Server} is
+     * updated by the {@link HealthCheckUpdateHandler}. This feature is enabled when the
+     * {@link HealthCheckService} is updatable.
+     *
+     * @see #updatable(boolean)
+     * @see #updatable(HealthCheckUpdateHandler)
+     */
+    public HealthCheckServiceBuilder updateListener(HealthCheckUpdateListener updateListener) {
+        if (updateHandler == null) {
+            throw new IllegalStateException("updating healthiness is disabled.");
```

Comment:
Could you specify why it's disabled?

Reply:
I guess a user might want to do this:

```java
builder.updateListener(...)
       .updatable(true);
```

.. which will raise an exception in `updateListener()`.

How about deferring the check to `build()`?

Reply:
Looks better.